core/src/components/checkbox/checkbox.tsx

```diff
@@ -253,11 +262,23 @@ export class Checkbox implements ComponentInterface {
 
     renderHiddenInput(true, el, name, checked ? value : '', disabled);
 
+    // Determine appropriate accessible name
+    const hasLabelContent = el.textContent !== '';
+    const inputLabelId = inputId + '-lbl';
+
+    // The host element must have a checkbox role to ensure proper VoiceOver
+    // support in Safari for accessibility
     return (
       <Host
+        role="checkbox"
         aria-checked={indeterminate ? 'mixed' : `${checked}`}
         aria-describedby={this.getHintTextID()}
         aria-invalid={this.getHintTextID() === this.errorTextId}
+        aria-labelledby={hasLabelContent ? inputLabelId : null}
+        aria-label={!hasLabelContent ? inheritedAttributes['aria-label'] || 'checkbox' : null}
```

Comment:
By using `inheritedAttributes['aria-label']` here, do we still need to [include](https://github.com/ionic-team/ionic-framework/blob/bbeff5b6f2356b69809c56924a1a803dcc95eefc/core/src/components/input/input.tsx#L821) it on `native-input`?

Reply:
I don't understand what you mean, I'm sorry

Reply:
Discussed offline, there shouldn't be any issues with leaving it as is. Plus, by removing it from `.native-input` then it might lead to changes that we didn't expect down the line.